src/portal/lib/src/service/quota.service.ts

```diff
@@ -0,0 +1,87 @@
+import { HttpClient, HttpResponse, HttpParams } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import {
+  HTTP_JSON_OPTIONS,
+  buildHttpRequestOptionsWithObserveResponse,
+} from "../utils";
+import {
+  QuotaSpec
+} from "./interface";
+import { map, catchError } from "rxjs/operators";
+import { Observable, throwError as observableThrowError } from "rxjs";
+import { Quota } from "./interface";
+/**
+ * Define the service methods to handle the replication (rule and job) related things.
+ *
+ **
+ * @abstract
+ * class QuotaService
+ */
+export abstract class QuotaService {
+  /**
+   *
+   * @abstract
+   * returns {(Observable<ReplicationRule[]>)}
+   *
+   * @memberOf QuotaService
+   */
+  abstract getQuotaList(page?, pageSize?, sortBy?: any):
+    any;
+
+  abstract updateQuota(
+    id: number,
+    rep: QuotaSpec
+  ): Observable<any>;
+}
+
+/**
+ * Implement default service for replication rule and job.
+ *
+ **
+ * class QuotaDefaultService
+ * extends {QuotaService}
+ */
+@Injectable()
+export class QuotaDefaultService extends QuotaService {
+
+  constructor(
+    private http: HttpClient
+  ) {
+    super();
+  }
+
+  public getQuotaList(page?, pageSize?, sortBy?: any):
+    any {
+    const sortByFiled = sortBy ? `&sort=${sortBy}` : '';
+
+    let params = new HttpParams();
+    if (page && pageSize) {
+      params = params.set('page', page + '').set('page_size', pageSize + '');
+    }
+
+
+
+    return this.http
+      .get<HttpResponse<Quota[]>>(`/api/quotas?reference=project${sortByFiled}`, buildHttpRequestOptionsWithObserveResponse(params))
+      .pipe(map(response => {
+        return response;
```

Comment:
`response` should exclude the quota which `ref` is empty.

Reply:
This situation has been dealt with.